<launch>

  <arg name="multi_robot"                 default="robot_5"/>
  <arg name="frame_id_global"             default="map"/>
  <arg name="frame_id_odom"               default="odom"/>
  <arg name="frame_id_base_footprint"     default="base_footprint"/>
  <arg name="topic_twist"                 default="rover_odo"/>
  <arg name="topic_initial_pose"          default="initialpose"/>
  <arg name="topic_odom"                  default="odom"/>
  <arg name="initial_pose_x"              default="0"/>
  <arg name="initial_pose_y"              default="0"/>
  <arg name="initial_pose_yaw"            default="0"/>

  <node pkg="rosbag" type="play" name="player" output="screen" args="/home/ros/ダウンロード/encoder.bag"/>

  <node pkg="imu_filter_madgwick" type="imu_filter_node" name="imu_filter_node">
    <remap from="imu/data_raw"          to="robot_5/robot_5/mynteye/imu/data_raw" />
    <param name="use_mag"               value="false" />
    <param name="fixed_frame"           value="$(arg frame_id_odom)" />
    <param name="publish_tf"            value="false"/>
    <!-- <param name="gain"                  value="0.02" /> -->
    <!-- <param name="remove_gravity_vector" value="true" /> -->
  </node>

  <node pkg="tf" type="static_transform_publisher" name="map_tf" args="0 0 0 0 0 0 $(arg frame_id_global) $(arg frame_id_odom) 60" />
  <node pkg="tf" type="static_transform_publisher" name="imu_tf" args="0 0 0 0 0 0 $(arg frame_id_base_footprint) camera_imu_frame 60" />

  <include file="$(find potbot_example)/launch/navigation/robot_localization.launch">
    <arg name="frame_id_global"         value="$(arg frame_id_global)"/>
    <arg name="frame_id_odom"           value="$(arg frame_id_odom)"/>
    <arg name="frame_id_base_footprint" value="$(arg frame_id_base_footprint)"/>
    <arg name="frame_id_world"          value="$(arg frame_id_odom)"/>
    <arg name="file_fusion"             value="$(find potbot_example)/config/megarover21/robot_localization.yaml"/>
  </include>

  <node pkg="potbot_example" type="twist_to_odometry" name="twist_to_odometry" output="screen">
    <param name="publish_tf"                value="false"/>
    <param name="frame_id_odom"             value="$(arg frame_id_odom)"/>
    <param name="frame_id_robot_base"       value="$(arg frame_id_base_footprint)"/>
    <param name="topic_sub_twist"           value="$(arg topic_twist)"/>
    <param name="topic_sub_initial_pose"    value="$(arg topic_initial_pose)"/>
    <param name="topic_pub_odom"            value="$(arg topic_odom)"/>
    <param name="initial_pose_x"            value="$(arg initial_pose_x)"/>
    <param name="initial_pose_y"            value="$(arg initial_pose_y)"/>
    <param name="initial_pose_yaw"          value="$(arg initial_pose_yaw)"/>
  </node>

</launch>
